### DISCUSSION TITLE
Can Macdriver use some kind of table or grid widget from Cocoa?

### DISCUSSION THREAD

**Discussion by ErnestoBorio** [progrium/darwinkit#68] [Q&A] [answered]
I need to have some sort of table or grid where I can access individual cells and scroll the whole widget.
Is anything like this usable from Macdriver?

I did work with Cocoa & Objective-C some time ago, but never used a table widget, not even sure how it's called.

My ideal would be a Cocoa widget. In the worst case scenario I could default to Webkit though.

**Comment by progrium** [comment]
macdriver is just wrapping all the cocoa apis. if you find what you need in official apple docs, you can use it with macdriver even if there isn't a type for it.

**Reply by ErnestoBorio** [reply]
Thanks! I can't mark this as the answer because the request returns an error.

**Comment by ErnestoBorio** [comment]
An example of a regular MacOS window would be great. I've run the `/examples` but wouldn't know how to implement a regular window to get started adding widgets to it.

**Comment by progrium** [comment]
like all of us you'd have to figure it out. there's also no way around reading the official apple docs because all the answers are there, you just have to translate to using the names from go.

**Comment by progrium** [comment, marked as answer]
if you get a window with widgets working, happy to add to examples
